sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientCache.scala

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.{CosmosAsyncClient, CosmosClientBuilder, ConsistencyLevel}
+import com.azure.cosmos.implementation.{CosmosClientMetadataCachesSnapshot, SparkBridgeInternal}
+import org.apache.spark.broadcast.Broadcast
+
+import scala.collection.concurrent.TrieMap
+
+private[spark] object CosmosClientCache {
+  private val cache = new TrieMap[CosmosClientConfiguration, CosmosAsyncClient]
+
+  private[spark] def apply(cosmosClientConfiguration: CosmosClientConfiguration,
```

Comment:
the class is already defined as private with spark scope, why do we need to explicitly set private spark scope for every method?

Reply:
Good catch